drivers/hdc1000/hdc1000.c

```diff
@@ -87,25 +91,31 @@ int hdc1000_init(hdc1000_t *dev, i2c_t i2c, uint8_t address)
     dev->addr = address;
     dev->initialized = false;
 
+    i2c_acquire(dev->i2c);
     /* initialize the I2C bus */
     if (i2c_init_master(i2c, I2C_SPEED) < 0) {
+        i2c_release(dev->i2c);
         return -1;
     }
+    i2c_release(dev->i2c);
 
     if (hdc1000_test(dev)) {
         return -2;
     }
 
+    i2c_acquire(dev->i2c);
```

Comment:
Is there a reason why you `aqcuired` i2c before generating `reg` here while you didn't do so for example in line 130?